src/App.tsx

```diff
@@ -1,223 +1,134 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  setIsLoadingContext,
+  functonsContext,
+} from './components/Context/context';
+import { Todo } from './types/Todo';
+import { Filter } from './components/Filter/Filter';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [allTodos, setAllTodos] = useState<Todo[] | null>(null);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [activeTodos, setActiveTodos] = useState<Todo[] | null>(null);
+  const [errorStatus, setErrorStatus] = useState('');
+  const [currentInput, setCurrentInput] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  const visibleTodos = useMemo(() => {
+    let result = allTodos;
+
+    if (allTodos) {
+      if (filter === 'active') {
+        result = allTodos.filter(todo => !todo.completed);
+      }
+
+      if (filter === 'completed') {
+        result = allTodos.filter(todo => todo.completed);
+      }
+    }
+
+    return result;
+  }, [filter, allTodos]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+  const setErrorWithTimer = useCallback((message: string) => {
+    setErrorStatus(message);
+    setTimeout(() => setErrorStatus(''), 3000);
+  }, []);
+
+  const loadUserTodos = useCallback(async () => {
+    if (user) {
+      const userTodos = await getTodos(user.id)
+        .catch(() => {
+          setErrorWithTimer('Unable to get a todo');
+
+          return [];
+        });
```

Comment:
maybe better try/catch block? you still will execute next code even with this catch call